### ISSUE TITLE
API GET invoices has a status filter WISH for that in API GET orders, proposals, shipments, ...

### ISSUE THREAD

**Issue by JonBendtsen** [Dolibarr/dolibarr#27145]
### Feature Request

In the API when looking for invoices there is a status filter where I can with words filter the invoice by status: "draft | unpaid | paid | cancelled".

I wish that would be available for orders, proposals, shipments, ... to create some consistency.

![dolibarr_API_invoices_GET](https://github.com/Dolibarr/dolibarr/assets/2551481/5addd3b2-3e1d-4236-b7c1-865d4548c53c)
![dolibarr_API_orders_GET](https://github.com/Dolibarr/dolibarr/assets/2551481/e660cf12-5465-454c-b0e1-59c08409eee7)
![dolibarr_API_proposals_GET](https://github.com/Dolibarr/dolibarr/assets/2551481/8ac76901-d400-477b-8654-83a676815d3d)
![dolibarr_API_shipments_GET](https://github.com/Dolibarr/dolibarr/assets/2551481/65b73af9-9a7e-49b9-86af-22c1538b8dc1)

### Use case

I'm using Dolibarr as a backend for my dance class registration system, and I need to parse all proposals, orders, invoices and shipments for a given project to create a participation list and how far along the process that registration is. I use the individual status of proposals, orders, invoices, shipments to follow how far along the registration is.

Currently I just get all orders, parse all of them and put their json into a different list. With invoices I don't have to do that, the list comes premade, but for all the others I have to make it myself.

### Suggested implementation

probably just copy what is done for invoices to orders, proposals and shipments.

**Comment by hansemschnokeloch** [comment]
Hi @eldy 

I would like to implement the status filtering on each relevant API endpoint with a PR.

Given that status filtering is only implemented on the invoice endpoint, what do you think of taking advantage of this to standardize everything by using the status names defined in Dolibarr classes for the API, 
i.e. `STATUS_DRAFT`, `STATUS_VALIDATED `, `STATUS_CLOSED` and `STATUS_ABANDONED`
 instead of the actual status currently used `draft`, `unpaid`, `paid`, `cancelled`?

This will only break backward compatibility for one API endpoint

**Comment by JonBendtsen** [comment]
@hansemschnokeloch  I appreciate your fast and enthusiastic attempt at fulfilling my wish. But - FYI I will most likely not end up using it anyway because while developing the participation list I have realized that the plan I had for how to get, modify and present the data separately will not work well, so I might as well get all the data in one go and then analyse them.

I created the feature request because I had not looked at invoices using the API before that day, so when I noticed that invoices had this option I thought I could use that.

I hope someone else will end up using your work :-)
